gpu/preconditioner/block_jacobi_kernels.cu

```diff
@@ -166,6 +166,210 @@ GKO_ENABLE_IMPLEMENTATION_SELECTION(select_apply, apply);
 using compiled_kernels = syn::compile_int_list<1, 13, 16, 32>;
 
 
+template <typename IndexType>
+__global__ void compare_nzcount_adjecent_rows(size_type m, IndexType *row,
+                                              bool *v)
+{
+    size_type tidx = blockDim.x * blockIdx.x + threadIdx.x;
+
+    if (tidx >= m) {
+        return;
+    }
+    IndexType nnzthis = row[tidx + 1] - row[tidx];
+    IndexType nnzlast = (tidx > 0) ? (row[tidx] - row[tidx - 1]) : 0;
+    v[tidx] = (nnzthis == nnzlast) ? 0 : 1;
```

Comment:
Though this is okay, I think the readability would be better if instead of 1/0 would be substituted by TRUE/FALSE.